pkg/core.go

```diff
@@ -1,6 +1,50 @@
 package datago
 
-import "context"
+import (
+	"context"
+	"sync/atomic"
+)
+
+type BufferedChan[T any] struct {
```

Comment:
native channels don't expose the waiting queue, this small struct is a way around that